packages/wrangler/src/__tests__/init.test.ts

```diff
@@ -2608,6 +2608,101 @@ describe("init", () => {
 			).rejects.toThrowError();
 		});
 
+		it("should not inlcude migrations in config file when none are necessary", async () => {
+			const mockData = {
+				id: "memory-crystal",
+				default_environment: {
+					environment: "test",
+					created_on: "1987-9-27",
+					modified_on: "1987-9-27",
+					script: {
+						id: "memory-crystal",
+						tag: "test-tag",
+						etag: "some-etag",
+						handlers: [],
+						modified_on: "1987-9-27",
+						created_on: "1987-9-27",
+						usage_model: "bundled",
+						compatibility_date: "1987-9-27",
+					},
+				},
+				environments: [],
+			};
+
+			setMockResponse(
+				`/accounts/:accountId/workers/services/:scriptName`,
+				"GET",
+				() => mockData
+			);
+			setMockResponse(
+				`/accounts/:accountId/workers/services/:scriptName/environments/:environment/bindings`,
+				"GET",
+				() => []
+			);
+			setMockResponse(
+				`/accounts/:accountId/workers/services/:scriptName/environments/:environment/routes`,
+				"GET",
+				() => []
+			);
+			setMockResponse(
+				`/accounts/:accountId/workers/services/:scriptName/environments/:environment`,
+				"GET",
+				() => mockServiceMetadata.default_environment
+			);
+			setMockResponse(
+				`/accounts/:accountId/workers/scripts/:scriptName/schedules`,
+				"GET",
+				() => {
+					return {
+						schedules: [],
+					};
+				}
+			);
```

Comment:
Follow up: I noticed that if these endpoints return anything other than an array, empty or otherwise, it is not handled well.

Reply:
Has this been resolved? Can we handle this better?

Reply:
Currently the API returns empty arrays if there is nothing. An API error itself is handled higher up the functionality, before it hits the config generator.